File: 7-kyu/an-english-twist-on-a-japanese-classic/javascript/solution.js
const shiritori = words => {
  if (words.length === 0)
    return [];
  let ret = [];
  let prev = words[0].slice(-1);
  if (words[0].length > 0)
    ret.push(words[0]);
  for (let i = 1; i < words.length; i++) {
    if (words[i].length === 0)
      break;
    if (words[i][0] === prev) {
      ret.push(words[i]);
    } else {
      break;
    }
    prev = words[i].slice(-1);
  }
  return ret;
};

File: 7-kyu/an-english-twist-on-a-japanese-classic/python/solution.py
def game(words):
    if not words:
        return []
    ret = []
    if len(words[0]) > 0:
        ret.append(words[0])
        prev = words[0][-1]
    for word in words[1:]:
        if len(word) == 0:
            break
        if word[0] == prev:
            ret.append(word)
        else:
            break
        prev = word[-1]
    return ret
